### PR TITLE
Updated the "Basic - Parallel execution" sample

### PR DESCRIPTION
**Pull request by Ark-kun**
Modernized the sample pipeline code.
/assign @gaoning777 @hongye-sun
/label area/components

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/pipelines/1110)

### CODE REVIEW

**Comment by Ark-kun** [comment]
The sample tests have passed. https://prow.k8s.io/pr-history/?org=kubeflow&repo=pipelines&pr=1110

File: samples/basic/parallel_join.py [hunk 1]
```diff
@@ -14,38 +14,44 @@
...
-import kfp.dsl as dsl
+import kfp
+from kfp import dsl
+
+def gcs_download_op(url):
+    return dsl.ContainerOp(
+        name='GCS - Download',
+        image='google/cloud-sdk:216.0.0',
+        command=['sh', '-c'],
+        arguments=['gsutil cat $0 | tee $1', url, '/tmp/results.txt'],
```
Comment (gaoning777) [inline samples/basic/parallel_join.py]: Could you explain how $0 and $1 get replaced with the url and 'tmp/..'?

Comment (Ark-kun) [reply samples/basic/parallel_join.py]: It's shell script, right?
The full command-line is 

```sh -c 'gsutil cat $0 | tee $1' url /tmp/results.txt```

`$0` and `$1` are the first and the second arguments to sh.

P.S. If we were running a shell script file, then you'd have to use `$1` and `$2`

**Pull request merged**